grails-app/views/dashboard/megamenu.gsp

```diff
@@ -99,33 +100,49 @@
 
 <script type="text/javascript">
   $(document).ready(function () {
-      const path = window.location.pathname
-      const menuConfigValues = $(".menu-config-value").toArray();
-      const matchingSection = menuConfigValues.find(it => it.value.includes(path))
+    const path = window.location.pathname
+    const menuConfigValues = $(".menu-config-value").toArray();
+    const urlPartsWithSectionConfig = "${(grailsApplication.config.openboxes.menuSectionsUrlParts as JSON)}";
+    const parsedUrlPartsWithSection = JSON.parse(urlPartsWithSectionConfig.replace(/&quot;/g,'"'));
 
-      // Assign active-section class to matched section
-      if (matchingSection) {
-        const matchingMenuSection = $("#" + matchingSection?.name).get(0);
-        const matchingMenuSectionCollapsable = $("#" + matchingSection?.name + "-collapsed").get(0);
-        if (matchingMenuSection) matchingMenuSection.classList.add('active-section');
-        if (matchingMenuSectionCollapsable) matchingMenuSectionCollapsable.classList.add('active-section');
+    const matchSection = () => {
+      // match the whole url
+      const exactMatch = menuConfigValues.find(it => it.value.includes(path))
+
+      if(exactMatch) {
+        return exactMatch;
       }
 
+      // check if current section exists in object with wrongly underlined urls
+      const sectionFromJson = Object.keys(parsedUrlPartsWithSection).find(sectionName => {
+        return !!parsedUrlPartsWithSection[sectionName].some(section => path.includes(section));
+      })
```

Comment:
Alternatively if we want to avoid this parsing of LinkedHasMap to JSON we can do everything using groovy.
By everything I mean finding the matched section based on `config.openboxes.menuSectionUrlParts`

So instead of doing it with javascript we can do it with groovy

```
      <g:set
        var="matchedSection"
        value="${grailsApplication.config.openboxes.menuSectionsUrlParts.find{ key, val -> val.any{ request?.forwardURI?.contains(it)} }}"
     />
      return menuConfigValues.find(it => it.name === "${matchedSection?.key}");
```

@alannadolny @kchelstowski  @awalkowiak  what do you think ?

Reply:
The solution that Kacper propsed, fixed the problem with replacing &quotes to ". 
So now it looks like: 
`   const urlPartsWithSection = $("#menuSectionUrlParts").val();
    const parsedUrlPartsWithSection = JSON.parse(urlPartsWithSection);`
I think it is much cleaner than doing it in groovy

Reply:
@drodzewicz I love that idea, but it depends on if we want Alan to spend some more time on this one